service/internal/unittest/default_node.go

```diff
@@ -0,0 +1,194 @@
+				},
+				Status: corev1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:   "Ready",
+							Status: "True",
+							LastHeartbeatTime: metav1.Time{
+								Time: time.Now().Add(-5 * time.Minute),
+							},
+							LastTransitionTime: metav1.Time{
+								Time: time.Now(),
+							},
+							Reason:  "KubeletReady",
+							Message: "kubelet is posting ready status",
+						},
+					},
+				},
+			},
+			{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Node",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "ip-10-0-5-132.eu-central-1.compute.internal",
+					Labels:      map[string]string{},
+					ClusterName: "",
+				},
+				Spec: corev1.NodeSpec{
+					ProviderID: "aws:///eu-central-1b/i-0448c486fa2eda084",
+				},
+				Status: corev1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:   "Ready",
+							Status: "True",
+							LastHeartbeatTime: metav1.Time{
+								Time: time.Now().Add(-5 * time.Minute),
+							},
+							LastTransitionTime: metav1.Time{
+								Time: time.Now(),
+							},
+							Reason:  "KubeletReady",
+							Message: "kubelet is posting ready status",
+						},
+					},
+				},
+			},
+			{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Node",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "ip-10-0-5-211.eu-central-1.compute.internal",
+					Labels: map[string]string{
+						"giantswarm.io/machine-deployment": "abc123",
+					},
+					ClusterName: "",
+				},
+				Spec: corev1.NodeSpec{
+					ProviderID: "aws:///eu-central-1b/i-0448c486fa2eda084",
+				},
+				Status: corev1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:   "Ready",
+							Status: "True",
+							LastHeartbeatTime: metav1.Time{
+								Time: time.Now().Add(-5 * time.Minute),
+							},
+							LastTransitionTime: metav1.Time{
+								Time: time.Now(),
+							},
+							Reason:  "KubeletReady",
+							Message: "kubelet is posting ready status",
+						},
+					},
+				},
+			},
+			{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "Node",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "ip-10-0-5-230.eu-central-1.compute.internal",
+					Labels: map[string]string{
+						"giantswarm.io/machine-deployment": "abc123",
+					},
+					ClusterName: "",
+				},
+				Spec: corev1.NodeSpec{
+					ProviderID: "aws:///eu-central-1b/i-0448c486fa2eda084",
+				},
+				Status: corev1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:   "Unknown",
+							Status: "Unknown",
+							LastHeartbeatTime: metav1.Time{
+								Time: time.Now().Add(-5 * time.Minute),
+							},
+							LastTransitionTime: metav1.Time{
+								Time: time.Now(),
+							},
+							Reason:  "KubeletNotReady",
+							Message: "kubelet is posting unknown status",
+						},
+					},
+				},
+			},
+		},
+	}
+	return nodes
+}
+
+func NewNode() corev1.Node {
```

Comment:
As an idea. What about having `NewMasterNode` and `NewWorkerNode` and in `DefaultNotes` we simply call these so that we can compose the list of notes using the constructors. I could see it is more modular and maintainable.

Reply:
I changed it a bit, I still need to create in the tests an additional node to check if the count works without caching and with caching.